### PR TITLE
Add collapsible container plugin

### PR DESCRIPTION
**Pull request by fantactuka**
Kind of an experiment with nested elements (vs going w/decorator and nested editors). Main part is in transformers that enforce nesting structure (and if it's violated unwrap content and convert collapsible content into regular paragraphs)
```
CollapsibleContainer
  > CollapsibleTitle
  > CollapsibleContent[isShadowRoot]
    > Paragraph
    > Heading
    > ...
```

https://user-images.githubusercontent.com/132642/192806800-292aa729-c5e6-41d1-9ba6-02259f1d9d6e.mov

### CODE REVIEW

File: packages/lexical-playground/src/plugins/ToolbarPlugin/index.tsx [hunk 1]
```diff
@@ -1121,6 +1122,14 @@ export default function ToolbarPlugin(): JSX.Element {
               <i className="icon sticky" />
               <span className="text">Sticky Note</span>
             </DropDownItem>
+            <DropDownItem
+              onClick={() => {
+                editor.dispatchCommand(INSERT_COLLAPSIBLE_COMMAND, undefined);
+              }}
+              className="item">
+              <i className="icon plus" />
```
Comment (thegreatercurve) [inline packages/lexical-playground/src/plugins/ToolbarPlugin/index.tsx:1130]: Maybe `arrow-right` instead?

**Review by thegreatercurve — COMMENTED** [review]
LGTM! Out of interest, how badly does this mess up browser search for the document? You could possibly look at adding something like `hidden=until-found` but it's still in early development: https://developer.chrome.com/articles/hidden-until-found

**Review by thegreatercurve — APPROVED** [review]

**Comment by Nirlah** [comment]
I like the nested nodes approach, it could be nice to have draggable blocks support inside the container and from/to the outside.

File: packages/lexical-playground/src/plugins/ToolbarPlugin/index.tsx (hunk 1 above)
Comment (acywatson) [reply packages/lexical-playground/src/plugins/ToolbarPlugin/index.tsx:1130]: +1

**Review by zurfyx — APPROVED** [review]
Awesome work! Not sure how to best validate the CollapsiblePlugin logic besides assuming that you came across them via trial and error and that we potentially still missed some that we will have to address over time.

**Comment by acywatson** [comment]
Never got a chance to finish this, but a while back I was messing around with doing this leaning more on browser native behaviors and semantic html:

https://codesandbox.io/s/lexical-plain-text-example-forked-jrrb38?file=/src/nodes/ToggleNode.js

Yours is certainly way better, just thought I would share

**Comment by fantactuka** [comment]
> Never got a chance to finish this, but a while back I was messing around with doing this leaning more on browser native behaviors and semantic html:
> [codesandbox.io/s/lexical-plain-text-example-forked-jrrb38?file=/src/nodes/ToggleNode.js](https://codesandbox.io/s/lexical-plain-text-example-forked-jrrb38?file=/src/nodes/ToggleNode.js)

I completely forgot about details/summary tags, thank you! It's actually a way better html structure as it both solves collab problem (collapsed/open state won't be synchronized) and auto-open if content matches CMD+F query

**Comment by acywatson** [comment]
> > Never got a chance to finish this, but a while back I was messing around with doing this leaning more on browser native behaviors and semantic html:
> > [codesandbox.io/s/lexical-plain-text-example-forked-jrrb38?file=/src/nodes/ToggleNode.js](https://codesandbox.io/s/lexical-plain-text-example-forked-jrrb38?file=/src/nodes/ToggleNode.js)
> 
> I completely forgot about details/summary tags, thank you! It's actually a way better html structure as it both solves collab problem (collapsed/open state won't be synchronized) and auto-open if content matches CMD+F query

**Pull request merged**